### PR TITLE
feat: Configurable Dekorate integration

### PR DESCRIPTION
**Pull request by manusa**
Relates to:
- fabric8io/fabric8-maven-plugin#1767
- fabric8io/fabric8-maven-plugin#1483

### CODE REVIEW

File: openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java [hunk 1]
```diff
@@ -260,6 +262,10 @@
     @Parameter(property = "jkube.resourceType")
     private ResourceFileType resourceFileType = yaml;
 
+    // When resource generation is delegated to Dekorate, should fabric8 resources be merged with Dekorate's
```
Comment (rohanKanojia) [inline openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java]: Shouldn't it be jkube here?

File: openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java [hunk 2]
```diff
@@ -338,6 +344,15 @@ public void executeInternal() throws MojoExecutionException, MojoFailureExceptio
         if (skipResource) {
             return;
         }
+        if (useDekorate(project) && mergeWithDekorate) {
+            log.info("Dekorate detected, merging fabric8 and Dekorate resources");
```
Comment (rohanKanojia) [inline openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java]: typo,

**Comment by manusa** [comment]
Once https://github.com/fabric8io/fabric8-maven-plugin/pull/1767 gets approved->merged;
- [ ] Port additional changes

**Comment by manusa** [comment]
All requested changes in https://github.com/fabric8io/fabric8-maven-plugin/pull/1767 have been ported.
PR is ready.

**Comment by rohanKanojia** [comment]
Thanks, I approved it during previous review too :laughing: . Shall we merge this then?

**Comment by manusa** [comment]
> Thanks, I approved it during previous review too  . Shall we merge this then?

Yes please, IT tests are green again after https://github.com/jshiftio/jkube-integration-tests/pull/9

**Pull request merged**